barretenberg/cpp/src/barretenberg/vm2/simulation/concrete_dbs.hpp

```diff
@@ -72,6 +77,14 @@ class MerkleDB final : public HighLevelMerkleDBInterface {
     // It's usually ok for mutexes but a gadget is big...
     PublicDataTreeCheckInterface& public_data_tree_check;
     NullifierTreeCheckInterface& nullifier_tree_check;
+
+    // Counters only in the HighLevel interface.
+    uint32_t nullifier_counter = 0;
+    uint32_t note_hash_counter = 0;
+    uint32_t l2_to_l1_msg_counter = 0;
+    // Set for semantics.
+    using Slot = FF;
+    std::set<Slot> storage_set;
```

Comment:
unordered_set unless you need to preserve order when visiting. 

(include `<unordered_set>`)